meetings/2021-11-23.md

```diff
@@ -0,0 +1,115 @@
+* ...: #213 renamed some properties related to grants and consents. It is now more intuitive when you read examples (turtle).
+* ...: We only have few things left on [the project board](https://github.com/solid/data-interoperability-panel/projects/1), one of them is about references list which we can dive into today.
+
+### Sai-js update
+
+* Pavlik: PR for authorization agent - https://github.com/janeirodigital/sai-js/pull/25
+
+* Pavlik: Hopefully people can look it over this week and ideally have it merged by end of the week.
+
+* Justin: I started going through that PR and I'm working on Java implementation.
+* ...: In the scenario where you have application that is a bot running on behalf or organization. The user has given consent to that application but the application has many users, since it works as service. The way that implementation class for application is written, it gets instantiated with WebID of the agent that is the current user. This was just eyeballing, I was trying to thing though scenario above. Weather this class needs to change or service needs to keep some state in context. When it wants to go and access data that it wants to access, than is in a state where it uses the application class as per user basis.
+* ...: I started thinking about how authN and authZ infrastructure looks like, with strongly identifiable authentication and the token that it provides to the Resource Server. And the different Authorization Services that are in the mix.
+
+* Pavlik: Great use case - haven't given it a ton of thought yet. From a resource standpoint we always expect to get the social agent and the client (application). Need to be careful how we expect those pairings to get pulled together. The scenario presented there are two cases: 
+- Server-side service used as an application by the user (e.g. user logs in and the user's webid is leveraged, and app needs to keep some state to know which to send for which user). Would be a different instance of fetch per user
+- Less explored - if the user is giving access to the organization the runs the application and the application is running "headless", when requests are made to the storage, the "organization" would appear as the social agent with the client identifier. Need to clarify who is being authorization - organization vs. user, and whose webid is presented.
+
+* Justin: I agree with how you laid out the two scenarios. It would be probably fine and appropriate to maintain the pairing as organization as social agent and application, when it operates in this 'headless' way. In the past the we were using same uri for webid and client_id. In application profile we already identify the responsible social agent as author.
+* ...: We have to detail out a little bit what the moving parts look on the authorization part on Solid-OIDC. Actually that's the first use case I need to demonstrate with Java library. I can take action item to dig into this one.
```

Comment:
```suggestion
* Justin: I agree with how you laid out the two scenarios. It would be probably fine and appropriate to maintain the pairing as organization as social agent and application, when it operates in this 'headless' way. In the past, the we were using same URI for WebID and `client_id`. In application profile, we already identify the responsible social agent as author.
  * We have to detail out a little bit what the moving parts look on the authorization part on Solid-OIDC. Actually that's the first use case I need to demonstrate with Java library. I can take action item to dig into this one.
```